app/views/unattended/provisioning_templates/snippet/remote_execution_ssh_keys.erb

```diff
@@ -58,6 +63,7 @@ EOF
   command -v restorecon && restorecon -RvF <%= ssh_path %> || true
 
 <% if ssh_user != 'root' && host_param('remote_execution_effective_user_method') == 'sudo' -%>
+$PKG_MANAGER_INSTALL sudo
```

Comment:
This can fail on systems with yum/dnf if the system is not entitled to consume content, even if the package is already installed. I'm not sure if that can actually happen in the wild, but could we stay on the safe side and only try installing it if it is missing?

Reply:
thats true. updated the PR accordingly.